src/resolve/resolved-dns-scope.c

```diff
@@ -411,6 +411,11 @@ static int dns_scope_socket(
                 }
         }
 
+        /* Set IP_RECVERR or IPV6_RECVERR to get ICMP error feedback. See discussion in #10345. */
+        r = setsockopt(fd, SOL_IP, sa.sa.sa_family == AF_INET ? IP_RECVERR : IPV6_RECVERR, &const_int_one, sizeof(const_int_one));
```

Comment:
we probably should do this only if `type == SOCK_DGRAM`? After all this only applies to datagram based transports afaics

Reply:
OK. Done.